opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -538,6 +538,21 @@ def use_span(
                 self.source._current_span_slot.set(  # pylint:disable=protected-access
                     span_snapshot
                 )
+
+        except Exception as error:  # pylint: disable=broad-except
+            if (
+                span.status is None
+                and span._set_status_on_exception  # pylint:disable=protected-access  # noqa
```

Comment:
Maybe you covered this in #297, but why not make `_set_status_on_exception` public? (Also, what's an example of a span where we want this to be false?)

Reply:
It can be made public, that would mean that we accept changing this behavior even after the span was created while specifically telling it to set or not its status on an exception being raised.

Maybe we want to handle the exception in the code that surrounds the span and not bother setting a status if the exception is raised? :man_shrugging: This was specifically [requested](https://github.com/open-telemetry/opentelemetry-python/issues/292#issuecomment-554443347) in the issue.

Reply:
That's something I missed on the last review. We should not be capturing the exception here!

```python
try:
    with tracer.start_as_current_span("foo") as span:
        raise ValueError()
except ValueError as ex:
    print("exception handling in the app")
else:
    print("uh oh")  # our context manager eats the error!
```

Reply:
Responding to the other comments:

> we accept changing this behavior even after the span was created

Or make it a read-only public property (which is easier in the SDK than the API).

> This was specifically requested in the issue.

I wonder if we want this to be a global setting instead of per-span. @Oberon00 what did you have in mind here?

Reply:
> That's something I missed on the last review. We should not be capturing the exception here!
> 
> ```python
> try:
>     with tracer.start_as_current_span("foo") as span:
>         raise ValueError()
> except ValueError as ex:
>     print("exception handling in the app")
> else:
>     print("uh oh")  # our context manager eats the error!
> ```

Good catch, I have committed a fix.

Reply:
Thanks, looks good!

Reply:
I did have the per-span setting as it is now in mind. But thinking about it again, explicitly setting the status to OK does the same thing, so this might not be necessary.

@c24t , @ocelotl: I was confused why we shouldn't capture (in my mind = record) the exception here, until I realized that you meant that we swallow the exception. 😄